### PR TITLE
DOC: why reformatting PRs will be rejected

### PR DESCRIPTION
**Pull request by mattip**
Make it clear that we do not invite reformatting PRs. Other suggestions for the message are welcome.

### CODE REVIEW

**Review by rossbar — APPROVED** [review]
Wording and rationale seem good to me, thanks @mattip

File: doc/source/dev/index.rst [hunk 1]
```diff
@@ -93,6 +93,15 @@ Here's the short summary, complete TOC links are below:
      changed behavior in docstrings, keeping to the NumPy docstring
      :ref:`standard<howto-document>`.
 
+  * NumPy is a project with a lot of technical debt. While we apprectiate
```
Comment (eric-wieser) [inline doc/source/dev/index.rst:96]:
```suggestion
  * NumPy is a project with a lot of technical debt. While we appreciate
```

File: doc/source/dev/index.rst [hunk 2]
```diff
@@ -93,6 +93,15 @@ Here's the short summary, complete TOC links are below:
      changed behavior in docstrings, keeping to the NumPy docstring
      :ref:`standard<howto-document>`.
 
+  * NumPy is a project with a lot of technical debt. While we appreciate
+    efforts to refactor and clean up code, please DO NOT submit a pull request
+    whose entire purpose is to convert formatting to f-strings, or reformat
```
Comment (eric-wieser) [inline doc/source/dev/index.rst:98]: Perhaps worth fitting the word `stylistic` in here somewhere. We might also want to mention this paragraph in the section that describes coding style too.
```suggestion
    whose entire purpose is stylistic, such as converting formatting to f-strings, or reformating
```

**Comment by charris** [comment]
At some point we will want to modernize, and f-strings are a definite improvement for readability. I've certainly done massive updates myself in the past, replacing `intp` by `npy_intp` for instance.

**Comment by mattip** [comment]
Closing, it seems @charris disagrees with the premise and has merged the PRs.

**Comment by charris** [comment]
> it seems @charris disagrees with the premise and has merged the PRs.

They were there, reviewed, and improvements, so in they went :) I can see not wanting hundreds of these all at once -- there are some 284 remaining uses of format -- but they aren't difficult to review. My memory may be a bit vague, but one of the arguments against these cleanups was that they changed the history, but that is in the nature of things and cannot be avoided in an ongoing project.

**Comment by eric-wieser** [comment]
I think as long as someone is willing to review them, and we trust that someone to look carefully enough to catch regressions like forgetting the `f`, I'm fine with the PRs. I think I share @mattip's opinion of not really wanting to review them myself.

If we do want to convert the remaining 284 uses, I'd much rather we just use an automated tool run by a trusted existing contributor to do so, and skip the manual review.

**Comment by charris** [comment]
> I'd much rather we just use an automated tool

My thought also. I'm waiting for someone somewhere to take the bait...

**Comment by charris** [comment]
And there are some 496 uses of `" % `. There was a python proposal to deprecate that usage, but it got walked back. Perhaps we should just encourage the use of f-strings in new code. For tests, I'm hoping for a test rewrite at some point that will modernize the code.

**Comment by mattip** [comment]
I wonder if there is a offshoot of mypy/black to spit out the AST, look for "string %" or "string.format" and would reformat the code.

**Comment by charris** [comment]
f-strings don't work everywhere, in particular as templates defined in a different context, so there is that. I wonder if any of the linting tools are clever enough to catch that?

**Pull request closed without merging**